neural_lam/weather_dataset.py

```diff
@@ -11,6 +11,7 @@
 
 # First-party
 from neural_lam.datastore.base import BaseDatastore
+from neural_lam.utils import check_time_overlap, get_time_step
 
 
 class WeatherDataset(torch.utils.data.Dataset):
```

Comment:
Implement the relevant changes on the WeatherDataset side that were required for succesfull experiments:
> Here is the exact changes that I made to neural-lam to make everything works despite issue 1 above: https://github.com/joeloskarsson/neural-lam-dev/commit/2f0e6a0137244d1073ced1db6165de1bca396be3